red_october/README.rst

```diff
@@ -0,0 +1,203 @@
+
+It also allows for the creation of vaults, management of users, and delegation of decryption rights to the vaults.
+
+Technical
+---------
+
+Crypto is implemented by providing a ``red_october`` type to ``ir.attachment``, which utilizes
+a remote Red October instance to seamlessly encrypt and decrypt data on the fly.
+
+It also provides some fields & form widgets to allow for similar operation to
+internal fields with seamless encryption:
+
+* `.fields.RedOctoberChar` - Acts like `odoo.fields.Char`
+
+Developer Implementation
+========================
+
+While this module does technically provide the ability to create standalone encrypted
+files, that is not feasible from a usability perspective.
+
+Instead, the preferred approach to allowing encryption is to use the provided fields
+in a model and subsequently add into the view.
+
+`Model Example:`
+.. code-block:: python
+
+    from odoo.addons.red_october.fields import RedOctoberChar
+
+    class MyModel(odoo.models):
+        _name = 'my.model'
+        encrypted_field = RedOctoberChar()
+
+`View Example:`
+.. code-block:: xml
+
+    <record id="my_model_view_form" model="ir.ui.view">
+        <field name="name">My Model Form View</field>
+        <field name="model">my.model</field>
+        <field name="arch" type="xml">
+            <form string="My Model">
+                <field name="encrypted_field" />
+            </tree>
+        </field>
+    </record>
+
+Installation
+============
+
+* A working Red October installation is required to use this module, which can easily be
+  launched using Docker: ``docker run laslabs/alpine-red-october``
+* Install the Python Red October library ``pip install git+https://github.com/LasLabs/python-red-october``
+
+Usage
+=====
+
+Create the Vault
+----------------
+
+* Activate Developer Mode
+* Go to `Settings => Crypto => Vaults` to create a new vault.
+  * Make sure to uncheck the `Verify Cert` option if your Red October server is using a self
+    signed cert.
+* Once created, a vault must be initialized with an administrative user using the
+  `Action => Init Vault` button from the vault form view.
+  * If the vault was already activated externally, make sure to check the
+    `Already Active` box.
+
+Crypto Profiles
+---------------
+
+Crypto Profiles are currently 1:1 with Odoo Users. If a user does not have a Crypto
+Profile, it will be created on demand using the Odoo password as the initial password.
+
+Crypto Menu
+-----------
+
+The Crypto Menu can be accessed in the top navigation bar by using the dropdown
+menu with a lock icon.
+
+The following actions are available in the Crypto Menu:
+* Change Profile Password
+* Delegate Decryption Rights
+
+Encryption/Decryption
+---------------------
+
+* Navigate to a form view implementing an encrypted field
+  * When in demo mode, a sample encrypted field is available under the `Preferences`
+    tab in the `res.users` form view
```

Comment:
I think the indent here is causing some rendering issues in the README. This appears quoted in the Github view.